test/printf.jl

```diff
@@ -6,21 +6,37 @@
 @test (@sprintf "%a" typemax(Int64)) == "0x7.fffffffffffffffp+60"
 @test (@sprintf "%A" typemax(Int64)) == "0X7.FFFFFFFFFFFFFFFP+60"
 
+#printing an int value
 for (fmt, val) in (("%i", "42"),
                    ("%u", "42"),
                    ("Test: %i", "Test: 42"),
                    ("%#x", "0x2a"),
                    ("%#o", "052"),
-                   ("%X", "2A"),
+                   ("%x", "2a"),
                    ("%X", "2A"),
                    ("% i", " 42"),
                    ("%+i", "+42"),
                    ("%4i", "  42"),
                    ("%-4i", "42  "),
-                   ("%a","0x2.ap+4"),
+                   ("%a", "0x2.ap+4"),
+                   ("%A", "0X2.AP+4"),
                    ("%20a","            0x2.ap+4"),
-                   ("%-20a","0x2.ap+4            "))
-    @test( @eval(@sprintf($fmt, 42) == $val))
+                   ("%-20a","0x2.ap+4            "),
+                   ("%f", "42.000000"),
+                   ("%g", "42")),
+     num in (UInt16(42), UInt32(42), UInt64(42), UInt128(42),
```

Comment:
It is supposed to be added in [Line 30](https://github.com/JuliaLang/julia/pull/15377/files#diff-bc7bacd08f3105fe45f402b53782d8feR30) (instead of `42`). I'll add it now!